### DISCUSSION TITLE
Add service for immediate charging

### DISCUSSION THREAD

**Discussion by Rantaki** [bimmerconnected/bimmer_connected#194] [Feature request]
**Is your feature request related to a problem? Please describe.**
During winter it is nice to heat up cabin before driving. Currently in BMW's hybrid models this happens using the high voltage battery of the car. The heating consumes a lot of energy and it is seen as a decrease in the electric range. For some reason, the car does not start charging again when the heating is started even though the battery is not full anymore.

**Describe the solution you'd like**
I would like to be able to create automation which triggers the immediate charging when ever heating is started. This way the energy needed in heating would mostly come from the mains instead of the battery. This way the battery would still be somewhat full when the driving starts.

**Describe alternatives you've considered**
This can be done manually, but not very elegant way. Basically this feature should be already in the car itself.

**Additional context**
![charge2](https://user-images.githubusercontent.com/16162066/93557598-0d28d300-f984-11ea-804c-3385ae26044a.jpg)

![charge1](https://user-images.githubusercontent.com/16162066/93557308-5a587500-f983-11ea-8390-fba35ada7dcc.jpg)

*NOTE: It might take me a while to respond on GitHub (I only maintain this in my free time), but I will respond.*

**Comment by gerard33** [comment]
From https://community.home-assistant.io/t/bmw-connecteddrive-component/39569/621?u=gerard33

> rantaki3
I need to correct my previous message. I did more research and it actually seems that charging is started when heating starts eating the battery. But I suspect the charging mode needs to be ”immediate” in this case.
Still it would be nice to see this feature in HA! I saw some work done on getting data of all trips out of the car (consumptions, speed, etc). This is something that even the BMW app cannot do, so that wouldnt it be awesome to have something like that?!

Just to make it clear: you are still looking for a service to activate immediate charging?

**Comment by Rantaki** [comment]
Hi,

It is not that mandatory anymore as my previous umdertanding of the functionality basically based on user error. So I am not actively waiting the feature, but I guess someone else might be.

Cheers

**Comment by MrJanD** [comment]
Hi,

I just got my new fully electric Mini Cooper SE. The App seems very similar to the BMW i3 App as well as the App @Rantaki previously posted some screenshots. Only _"Charge immediately"_ and _"Charge with off-peak tariff"_ can be selected. 
However, as you can imagine, BEV drivers want to maximize the life of their lithium-ion battery. Therefore, it is a good idea to not fully charge the battery. Since this is not an option in the BMW or Mini App, I would like to implement such a charging profile myself. This can be achieved by selecting _"Charge immediately"_ and than _"Charge with off-peak tariff"_ once the battery level reached 80%. 
Therefore, it would be really helpful to have the possibility to select _"Charge immediately"_ and _"Charge with off-peak tariff"_. Is this planned in the near future or is there a way I can help with this implementation?

**Comment by rikroe** [comment]
It seems that there are some services (see https://github.com/TA2k/ioBroker.bmw/blob/54a47a4bae6f97c6380a97d64734bb3a17d7fec6/main.js#L211-L212), altough I don't know what *off-peak tariff* is translated to.
